src/utils.js

```diff
@@ -343,3 +348,42 @@ const sleep = (millis) => {
 export const publicUtils = {
     sleep,
 };
+
+/**
+ * Returns a promise that resolves to an array of urls parsed from the resource
+ * available at the provided url.
+ *
+ * @param {String} url
+ * @param {String} encoding
+ * @param {Number} offset
+ * @param {Number} limit
+ * @param {Boolean} skipDuplicates
+ * @param {RegExp} urlRegExp
+ * @returns {Promise<Array>}
+ */
+export const downloadListOfUrls = ({ url, encoding, offset, limit, skipDuplicates, urlRegExp }) => {
+    if (!url) return Promise.reject(new Error('Resource URL must be provided.'));
+    encoding = encoding || 'utf8';
```

Comment:
You can define defaults above ^. `({ url, encoding = 'utf8', ... })`

Reply:
I know, I did that and then I remembered that `null` is a legitimate value when you define default parameter values, so I had to go with `encoding || 'utf8'` not to get a Buffer back from `requestPromise` on `null` encoding.